server/engine/Game.js

```diff
@@ -0,0 +1,139 @@
+const wordArray = require('./Words');
+
+class Game {
+  GUESS_POINT = 200;
+  CLUE_POINT = 100;
+
+  /**
+   * @param {array} players : game players array
+   */
+  constructor(players) {
+    this.word = '';
+    this.round = 0;
+    this.wordArray = [];
+    this.players = players;
+    this.maxRound = this.players.length * 2;
+
+    this.initGame();
+  }
+
+  /**
+   * Init & Restart Game
+   */
+  initGame() {
+    console.log(this.players)
+    // Get Array Data
+    this.wordArray = this.initWords();
+    this.players[0].isGuesser = true;
+    this.word = this.wordArray[0];
+    this.round = 0;
+  }
+
+  /**
+   * Get Words Array for game
+   */
+  initWords() {
+    return this.shuffle(wordArray).slice(0, this.maxRound);
+  }
+  
+  /**
+   * init Players Roles
+   */
+  initPlayerRoles() {
+    this.players.map((player) => {
+       player.isGuesser = false;
+       player.isGiver = false;
+    });
+  }
+
+  /**
+   * Check Guesser's Answer
+   * @param {string} answer
+   */
+  checkAnswer(answer) {
+    return this.word === answer ? true : false;
+  }
+
+  /**
+   * Go to Next Round
+   */
+  nextRound() {
+    this.round = this.round + 1;
+    this.initPlayerRoles();
+    this.setNextGuesser(this.round);
```

Comment:
What about the giver roles?

Reply:
Player who give a clue to a guesser. If other players' clues are duplicated, they can't be a giver. If answer is correct, only a guesser and a giver can get points in a round.